mitosheet/mitosheet/mito_dash/v1/spreadsheet.py

```diff
@@ -52,60 +49,86 @@ def analysis(self) -> RunnableAnalysis:
     from dash import Input, Output, callback, State
 
 
     class Spreadsheet(Component):
+        
+        # See documentation in the get_instance method
+        instances: Dict[str, Tuple[Any, Dict[str, Any]]] = dict()
 
         _children_props: List[str] = []
         _base_nodes = ['children']
         _namespace = 'dash_spreadsheet_v1'
         _type = 'MitoDashWrapper'
-        _prop_names = ['id', 'all_json', 'data', 'import_folder', 'spreadsheet_result', 'spreadsheet_selection']
+        _prop_names = ['id', 'all_json', 'data', 'import_folder', 'spreadsheet_result', 'spreadsheet_selection', 'mito_id', 'session_key']
         _valid_wildcard_attributes: List[str] = []
         available_properties = ['id', 'all_json', 'data', 'import_folder']
         available_wildcard_properties: List[str] = []
 
         def __init__(
-                self, 
-                *args: Union[pd.DataFrame, str, None],
-                id: str,
-                import_folder: Optional[str]=None,
-                code_options: Optional[CodeOptions]=None,
-                df_names: Optional[List[str]]=None,
-                sheet_functions: Optional[List[Callable]]=None, 
-                importers: Optional[List[Callable]]=None,
-                editors: Optional[List[Callable]]=None,
-                theme: Optional[MitoTheme]=None,
-                track_selection: bool=False,
-                
-        ):     
-            self.mito_id = id
+            self, 
+            *args: Union[pd.DataFrame, str, None],
+            id: Dict[str, str],
+            import_folder: Optional[str]=None,
+            code_options: Optional[CodeOptions]=None,
+            df_names: Optional[List[str]]=None,
+            sheet_functions: Optional[List[Callable]]=None, 
+            importers: Optional[List[Callable]]=None,
+            editors: Optional[List[Callable]]=None,
+            theme: Optional[MitoTheme]=None,
+            track_selection: bool=False,
+            mito_frontend_key: Optional[str]=None
+        ):    
+            
+            # First, we check that the user has passed an ID in the appropriate format
+            # as this is required for the component to work. It must be a dict with the 
+            # keys type and id, where type == ID_TYPE and id is a unique string
+            if not isinstance(id, dict):
+                raise Exception("You must pass an ID to the Spreadsheet component as a dict with the keys type and id")
+            if 'type' not in id:
+                raise Exception("You must pass an ID to the Spreadsheet component as a dict with the keys type and id")
+            if 'id' not in id:
+                raise Exception("You must pass an ID to the Spreadsheet component as a dict with the keys type and id")
+            if id['type'] != ID_TYPE:
+                raise Exception("You must pass an ID to the Spreadsheet component as a dict with the keys type and id")
```

Comment:
These are all the same error, can we condense it into: 

`if 'type' not in id or 'id not in id...`